### ISSUE TITLE
feat: support dialog behaviours

### ISSUE THREAD

**Issue by jlp-craigmorten** [guidepup/virtual-screen-reader#47]
See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role#possible_effects_on_user_agents_and_assistive_technology

> When the dialog is correctly labeled and focus is moved to an element (often an interactive element, such as a button) inside the dialog, screen readers should announce the dialog's accessible role, name and optionally description, along with announcing the focused element.

Although this is not normative, but MDN's opinion (with a disclaimer saying as such), testing with NVDA and VoiceOver this appears to be the case and seems not unreasonable to implement.

_Originally posted by @jlp-craigmorten in https://github.com/guidepup/virtual-screen-reader/issues/12#issuecomment-1930917405_

**Comment by cmorten** [comment]
Have commented in https://github.com/w3c/aria/issues/1195#issuecomment-1931550211 where the associated PR https://github.com/w3c/aria/pull/2054 adds clarity that this _will_ be a **SHOULD** expectation on assistive technology.

As the draft stands:

> Assistive technology SHOULD give precedence to exposing author defined dialog accessible descriptions when a dialog is invoked and user focus is moved to a descendant of the dialog element.
